cpmpy/solvers/gurobi.py

```diff
@@ -0,0 +1,376 @@
+    tpl_model: object, TEMPLATE's model object
+    _varmap: dict(), maps cpmpy variables to native solver variables
+    """
+
+    @staticmethod
+    def supported():
+        # try to import the package
+        try:
+            import gurobipy as gp
+            from datetime import datetime
+            valid_until = gp.Model().LicenseExpiration
+            today = datetime.today()
+            return today.year * 1e4 + today.month * 1e2 + today.day <= valid_until
+        except ImportError as e:
+            return False
+
+    def __init__(self, cpm_model=None, subsolver=None):
+        """
+        Constructor of the native solver object
+
+        Arguments:
+        - cpm_model: a CPMpy Model()
+        """
+        if not self.supported():
+            raise Exception(
+                "CPM_gurobi: Install the python package 'gurobipy' and make sure your licence is activated!")
+        import gurobipy as gp
+
+        # initialise the native solver object
+        self.env = gp.Env()
+        self.env.setParam("LogToConsole", 0)
+        self.env.setParam("OutputFlag", 0)
+        self.env.start()
+
+        self.gbi_model = gp.Model(env=self.env)
+        self._objective_value = None
+
+        # initialise everything else and post the constraints/objective
+        # it is sufficient to implement __add__() and minimize/maximize() below
+        super().__init__(name="gurobi", cpm_model=cpm_model)
+
+    def __add__(self, cpm_con):
+        """
+        Post a (list of) CPMpy constraints(=expressions) to the solver
+
+        Note that we don't store the constraints in a cpm_model,
+        we first transform the constraints into primitive constraints,
+        then post those primitive constraints directly to the native solver
+
+        :param cpm_con CPMpy constraint, or list thereof
+        :type cpm_con (list of) Expression(s)
+        """
+        # add new user vars to the set
+        self.user_vars.update(set(get_variables(cpm_con)))
+
+        # apply transformations, then post internally
+        # XXX chose the transformations your solver needs, see cpmpy/transformations/
+
+        cpm_cons = flatten_constraint(cpm_con)
+        cpm_cons = linearize_constraint(cpm_cons)
+
+        for con in cpm_cons:
+            self._post_constraint(con)
+
+        return self
+
+    def solve(self, time_limit=None, solution_callback=None, **kwargs):
+        """
+            Call the gurobi solver
+
+            Arguments:
+            - time_limit:  maximum solve time in seconds (float, optional)
+            - kwargs:      any keyword argument, sets parameters of solver object
+
+            Arguments that correspond to solver parameters:
+            <Please document key solver arguments that the user might wish to change
+             for example: log_output=True, var_ordering=3, num_cores=8, ...>
+            <Add link to documentation of all solver parameters>
```

Comment:
https://www.gurobi.com/documentation/9.5/refman/mip_models.html for the accessible version, maybe take some of them and list them in the line above as suggested

https://www.gurobi.com/documentation/9.5/refman/parameters.html#sec:Parameters for the full list